### DISCUSSION TITLE
Swap New Window / New Tab commands? (custom keymap.json)

### DISCUSSION THREAD

**Discussion by luckman212** [zed-industries/zed#6531] [Feature Requests]
New to Zed (well, I guess everyone is...). Using 0.79.1 on macOS 13.3.

I'm trying to set up a custom `keymap.json` to invert the default keybindings for <kbd>⌘ N</kbd> and <kbd>⌘ ⇧ N</kbd>.
I'd like <kbd>⌘ N</kbd> to open a new **Window** and <kbd>⌘ ⇧ N</kbd> to open a new **Tab**.

Here's the `keymap.json` I created:

```json
[
    {
        "bindings": {
            "cmd-n": "workspace::NewWindow",
            "cmd-shift-n": "workspace::NewFile"
        }
    }
]
```

But it has no effect. Any idea why?

**Comment by luckman212** [comment]
I should add that I updated to Zed 0.79.1 and still can't get this to function the way I want. It seems to ignore the custom keymap.json. Anyone know what I'm doing wrong here?

**Comment by JosephTLyons** [comment]
Hey @luckman212 - what happens if you remove the trailing `,` in `"cmd-shift-n": "workspace::NewFile",`?

**Reply by luckman212** [reply]
@JosephTLyons Nothing different. Actually my real keymap.json had no stray comma. That was just a typo or something that slipped into the GitHub comment. I still can't make this work. Does it work for you?

**Comment by krstp** [comment]
Kind of the same here... trying to use my own keymap and per below `Editor` and `Workspace` segment work as desired, but `Pane` for some odd reason does not, see below:
```
[
    {
        "context": "Editor",
        "bindings": {
            "ctrl-f12": "editor::FindAllReferences",
            "cmd-k left": "workspace::ActivatePreviousPane",
            "cmd-k right": "workspace::ActivateNextPane"
        }
    },
    {
        "context": "Workspace",
        "bindings": {
            "alt-cmd-1": [
                "workspace::ActivatePane",
                0
            ],
            "alt-cmd-2": [
                "workspace::ActivatePane",
                1
            ],
            "alt-cmd-3": [
                "workspace::ActivatePane",
                2
            ],
            "alt-cmd-4": [
                "workspace::ActivatePane",
                3
            ],
            "alt-cmd-5": [
                "workspace::ActivatePane",
                4
            ],
            "alt-cmd-6": [
                "workspace::ActivatePane",
                5
            ],
            "alt-cmd-7": [
                "workspace::ActivatePane",
                6
            ],
            "alt-cmd-8": [
                "workspace::ActivatePane",
                7
            ],
            "alt-cmd-9": [
                "workspace::ActivatePane",
                8
            ]
        }
    },
    {
        "context": "Pane",
        "bindings": {
            "cmd-k cmd-up": "pane::SplitUp",
            "cmd-k cmd-down": "pane::SplitDown",
            "cmd-k cmd-left": "pane::SplitLeft",
            "cmd-k cmd-right": "pane::SplitRight"
        }
    }
]
```
However...
- `CMD-K CMD-UP` and `CMK-K CMD-DOWN` works as desired, but it seems the same as `CMD-K UP` and `CMD-K DOWN` (`default` Atom bindings) 🤔 which I don't think is right as `defaults` should be overwritten.
- `CMD-K CMD-LEFT` is the same as `CMD-K LEFT` :(
- `CMD-K CMD-RIGHT` is the same as `CMD-K RIGHT` :(

**Reply by krstp** [reply]
I can see there are some `default` bindings that take precedence:
```
// Bindings from Sublime Text
[...]
    {
        "bindings": {
            "cmd-k cmd-left": "workspace::ActivatePreviousPane",
            "cmd-k cmd-right": "workspace::ActivateNextPane"
        }
    },
```

**Reply by krstp** [reply]
And more `defaults` from Atom:
```
    // Bindings from Atom
    {
        "context": "Pane",
        "bindings": {
            "cmd-k up": "pane::SplitUp",
            "cmd-k down": "pane::SplitDown",
            "cmd-k left": "pane::SplitLeft",
            "cmd-k right": "pane::SplitRight"
        }
    },
```

**Reply by krstp** [reply]
Also note that `CMD+ALT+3` is the same as `CMD+3` and this applies to all the numbers. So for some reason `defaults` do not get exactly overwritten.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
